Master (#1)
invenia/MaximumGeneratorProfitTCRDD: README.md

# MaximumGeneratorProfitTCRDD

> 

Reviewer: The README needs a bit of reformatting. Ideally each sentence would be on its own line.

The package allows the user to find the bid which maximises the profit of the Generator at
the slack bus using the Transmission Constrained Residual Demand Derivative (TCRDD).

> Two Main Functions compose the package, (A) maxGenProfit_tcrdd and (B) plots_TCRDD. 
> Function A finds the bid which maximises the profit, and function B creates a series of 
> plots to facilitate the analisys of the algorithm.

Reviewer: ```suggestion
Two Main Functions compose the package:

1. maxGenProfit_tcrdd: finds the bid which maximises the profit
2. plots_TCRDD: creates a series of 
plots to facilitate the analysis of the algorithm
```

### The Algorithm

> The algorithm is composed by two parts, (1) the Local Screening Loop (LSL) and (2) the 
> Bisection Loop (BL). The screening loop uses the initial bid0 and creates an aproximation
> of the profit function. Using this aproximation, it identifies the range [bid lower 

Reviewer: Might need a bit of re-working with line breaks in sentences here.
```suggestion
The algorithm is composed by two parts,

 1. Local Screening Loop (LSL)
 2. Bisection Loop (BL)
 
 The screening loop uses the initial bid0 and creates an approximation of the profit function. Using this approximation, it identifies the range [bid lower 
```

> (bid_lo) and bid higher (bid_hi)] in which the bid that maximises the generator profit 
> could be. Once the lower and upper bid range are identified, the bisection loop will find 

Reviewer: ```suggestion
(bid_lo) and bid higher (bid_hi)] which maximizes the generator profitability.
Once the lower and upper bid range are identified, the bisection loop will find 
```

> try to find if the approximations of the lower and upper bid intersect at some point. If 
> they do, the intersection bid is used to find a closer upper and lower range, or identify 

Reviewer: ```suggestion
try to find if the approximations of the lower and upper bid intersect at some point. 
If found, the intersection bid is used to find a closer upper and lower range, or identify 
```

> if the optimum has been found. If they dont intersect then, traditional bisection is done 

Reviewer: ```suggestion
if the optimum has been found. If they don't intersect then, traditional bisection is done 
```
